src/replication.c

```diff
@@ -487,10 +665,19 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd)
 
 	MTX_LOCK(&spec->rq_mtx);
 
-	spec->replica_count++;
 	spec->degraded_rcount++;
 	TAILQ_REMOVE(&spec->rwaitq, replica, r_waitnext);
 	TAILQ_INSERT_TAIL(&spec->rq, replica, r_next);
+
+	MTX_UNLOCK(&spec->rq_mtx);
+
+	/*
+	 * Sleep for timeout sec so that outstanding IOs got committed to disk
+	 */
+	sleep(spec->replica_io_timeout);
```

Comment:
thread should not be made to 'sleep' explicitly. All we need to make sure is: rebuild should not be started for this amount of time and IOs should be started on this replica.

Reply:
yes.. timestamp that need to check for next rebuild is good to have. This can also make code not to piggyback on ask_replica_status_all function, but, can have its own if needed.